examples/parallel_environment.py

```diff
@@ -0,0 +1,282 @@
+import gym
+
+gym.logger.set_level(40)
+
+from examples.argument_parser import default_argument_parser
+from smarts.core.agent import Agent, AgentSpec
+from smarts.core.agent_interface import AgentInterface
+from smarts.core.controllers import ActionSpaceType
+from smarts.core.sensors import Observation
+from smarts.env.hiway_env import HiWayEnv
+from smarts.env.wrappers.parallel_env import ParallelEnv
+from smarts.env.wrappers.frame_stack import FrameStack
+from typing import Dict, Tuple, Sequence
+
+
+class ChaseViaPointsAgent(Agent):
+    def act(self, obs: Sequence[Observation]) -> Tuple[float, int]:
+        # Here, we only utilise the newest frame from the stacked observations.
+        obs = obs[-1]
+
+        if (
+            len(obs.via_data.near_via_points) < 1
+            or obs.ego_vehicle_state.edge_id != obs.via_data.near_via_points[0].edge_id
+        ):
+            return (obs.waypoint_paths[0][0].speed_limit, 0)
+
+        nearest = obs.via_data.near_via_points[0]
+        if nearest.lane_index == obs.ego_vehicle_state.lane_index:
+            return (nearest.required_speed, 0)
+
+        return (
+            nearest.required_speed,
+            1 if nearest.lane_index > obs.ego_vehicle_state.lane_index else -1,
+        )
```

Comment:
I think we should use as primitive agent as possible.

```suggestion
class ChaseViaPointsAgent(Agent):
    def act(self, obs: Sequence[Observation]) -> Tuple[float, int]:
        # Here, we only utilise the newest frame from the stacked observations.
        newest_obs = obs[-1]
        speed_limit = newest_obs.waypoint_paths[0][0].speed_limit
        return (speed_limit, 0)
```

Reply:
Simplified the agent.